scipy/fft/tests/test_real_transforms.py

```diff
@@ -14,37 +20,41 @@
 # fftpack/test_real_transforms.py
 
 
+@skip_if_array_api_gpu
+@array_api_compatible
 @pytest.mark.parametrize("forward, backward", [(dct, idct), (dst, idst)])
 @pytest.mark.parametrize("type", [1, 2, 3, 4])
 @pytest.mark.parametrize("n", [2, 3, 4, 5, 10, 16])
 @pytest.mark.parametrize("axis", [0, 1])
 @pytest.mark.parametrize("norm", [None, 'backward', 'ortho', 'forward'])
 @pytest.mark.parametrize("orthogonalize", [False, True])
-def test_identity_1d(forward, backward, type, n, axis, norm, orthogonalize):
+def test_identity_1d(forward, backward, type, n, axis, norm, orthogonalize,
+                     xp):
     # Test the identity f^-1(f(x)) == x
-    x = np.random.rand(n, n)
+    x = xp.asarray(np.random.rand(n, n))
 
     y = forward(x, type, axis=axis, norm=norm, orthogonalize=orthogonalize)
     z = backward(y, type, axis=axis, norm=norm, orthogonalize=orthogonalize)
-    assert_allclose(z, x)
+    assert_close(z, x)
 
     pad = [(0, 0)] * 2
     pad[axis] = (0, 4)
 
-    y2 = np.pad(y, pad, mode='edge')
+    y2 = xp.asarray(np.pad(np.asarray(y), pad, mode='edge'))
     z2 = backward(y2, type, n, axis, norm, orthogonalize=orthogonalize)
-    assert_allclose(z2, x)
+    assert_close(z2, x)
 
 
+@skip_if_array_api
```

Comment:
I didn't see any failures when I removed this skip locally--is the idea to skip because it contributes nothing to array API testing rather than it fails?

I think I saw a discussion about the skipping philosophy somewhere else, and maybe this kind of debate came up. I wonder if a `pytest`-style `reason` keyword might eventually be helpful to annotate skips to help devs avoid having to read the test code to understand why the skip is there.

Reply:
Yes, I have used a lot of skips like this to mean that we only want to run the test for `numpy`.

In order to be able to do so with the flag set, rather than run the tests separately without the flag set, I have proposed that we change to this behaviour, or include it as a concise option in broader changes, in gh-19181.

At the minute, we get coverage by running `python dev.py test -s fft -b all` and `python dev.py test -s fft`, but it would be nice to just have to run the former.

>  I wonder if a pytest-style reason keyword might eventually be helpful to annotate skips to help devs avoid having to read the test code to understand why the skip is there.

Yep this sounds useful, feel free to mention that at some point on gh-19181.

Reply:
I suggest to delete all these skips right now. It makes the diff smaller and easier to understand, and it makes the  set of tests that are run with `-b all` a superset of the regular set of tests. Having a single test command to run will also make it easier to finalize this PR.